mmdet3d/core/points/base_points.py

```diff
@@ -153,28 +155,13 @@ def rotate(self, rotation, axis=None):
             axis = self.rotation_axis
 
         if rotation.numel() == 1:
-            rot_sin = torch.sin(rotation)
-            rot_cos = torch.cos(rotation)
-            if axis == 1:
-                rot_mat_T = rotation.new_tensor([[rot_cos, 0, -rot_sin],
-                                                 [0, 1, 0],
-                                                 [rot_sin, 0, rot_cos]])
-            elif axis == 2 or axis == -1:
-                rot_mat_T = rotation.new_tensor([[rot_cos, -rot_sin, 0],
-                                                 [rot_sin, rot_cos, 0],
-                                                 [0, 0, 1]])
-            elif axis == 0:
-                rot_mat_T = rotation.new_tensor([[0, rot_cos, -rot_sin],
-                                                 [0, rot_sin, rot_cos],
-                                                 [1, 0, 0]])
-            else:
-                raise ValueError('axis should in range')
-            rot_mat_T = rot_mat_T.T
-        elif rotation.numel() == 9:
-            rot_mat_T = rotation
+            rotated_points, rot_mat_T = rotation_3d_in_axis(
+                self.tensor[:, :3][None], rotation, axis=axis, return_mat=True)
+            self.tensor[:, :3] = rotated_points.squeeze(0)
+            rot_mat_T = rot_mat_T.squeeze(0)
         else:
-            raise NotImplementedError
-        self.tensor[:, :3] = self.tensor[:, :3] @ rot_mat_T
```

Comment:
elif rotation.numel() == 9:
      xxx
else:
      raise NotImplementedError ?

Reply:
See assertion on line 151. Rotation.numel must be 1 or 9

Reply:
Nevertheless, I added a comment